tools/deploy.go

```diff
@@ -0,0 +1,711 @@
+	}
+	defer revertFunc()
+
+	execCommand(kustomizePath, "create", "--resources=../../base,../../namespace", "--namespace=baremetal-operator-system")
+
+	if deployBasicAuthFlag {
+		execCommand(kustomizePath, "edit", "add", "component", "../../components/basic-auth")
+		execCommand(kustomizePath, "edit", "add", "secret", "ironic-credentials", "--from-file=username=ironic-username", "--from-file=password=ironic-password")
+	}
+
+	if deployTLSFlag {
+		execCommand(kustomizePath, "edit", "add", "component", "../../components/tls")
+	}
+}
+
+// helper function to read lines from a file.
+func readLines(filePath string) ([]string, error) {
+	file, err := os.Open(filePath)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	var lines []string
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		lines = append(lines, scanner.Text())
+	}
+
+	return lines, scanner.Err()
+}
+
+// helper function to write lines to a file.
+func writeLines(lines []string, filePath string) error {
+	file, err := os.Create(filePath)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	w := bufio.NewWriter(file)
+	for _, line := range lines {
+		fmt.Fprintln(w, line)
+	}
+
+	return w.Flush()
+}
+
+func updateOrAppendKeyValueInFile(filePath, key, value string) error {
+	lines, err := readLines(filePath)
+	if err != nil {
+		return fmt.Errorf("failed to read lines from file %s: %v", filePath, err)
+	}
+
+	updatedLines := []string{}
+	found := false
+	for _, line := range lines {
+		if strings.HasPrefix(line, key+"=") {
+			updatedLines = append(updatedLines, fmt.Sprintf("%s=%s", key, value))
+			found = true
+		} else {
+			updatedLines = append(updatedLines, line)
+		}
+	}
+	if !found {
+		updatedLines = append(updatedLines, fmt.Sprintf("%s=%s", key, value))
+	}
+
+	err = writeLines(updatedLines, filePath)
+	if err != nil {
+		return fmt.Errorf("failed to write updated lines to file %s: %v", filePath, err)
+	}
+
+	return nil
+}
+
+func splitYAMLDocuments(yamlContent []byte) [][]byte {
+	var documents [][]byte
+	var currentDoc []byte
+	lines := bytes.Split(yamlContent, []byte("\n"))
+
+	for _, line := range lines {
+		trimmedLine := bytes.TrimSpace(line)
+		if bytes.Equal(trimmedLine, []byte("---")) && len(trimmedLine) == 3 {
+			if len(currentDoc) > 0 {
+				documents = append(documents, append([]byte(nil), currentDoc...))
+				currentDoc = []byte{}
+			}
+		} else {
+			currentDoc = append(currentDoc, line...)
+			currentDoc = append(currentDoc, '\n')
+		}
+	}
+	if len(currentDoc) > 0 {
+		documents = append(documents, currentDoc)
+	}
+	return documents
+}
+
+// getKubeconfigPath determines the kubeconfig path from KUBECTL_ARGS, KUBECONFIG_PATH, or defaults to ~/.kube/config.
+func getKubeconfigPath() string {
+	kubectlArgs := os.Getenv("KUBECTL_ARGS")
+	kubeconfigPath := ""
+
+	kubeconfigPrefix := "--kubeconfig="
+	if strings.Contains(kubectlArgs, kubeconfigPrefix) {
+		if strings.Contains(kubectlArgs, " ") {
```

Comment:
I'd like to see whitelisting rather than blacklisting.  For example, check it matches `--kubeconfig=[a-zA-Z/.]+` and then verify its a file and its readable.